R/methods-epi_df.R

```diff
@@ -80,13 +80,39 @@ summary.epi_df = function(object, ...) {
   } 
   
   cn <- names(res)
-  nr <- vctrs::vec_size(x) 
-  not_epi_df <- (!("time_value" %in% cn) || !("geo_value" %in% cn) || vctrs::vec_size(res) > nr || any(i > nr))
   
-  if (not_epi_df) return(tibble::as_tibble(res))
+  # Duplicate columns, Abort
+  dup_col_names = cn[duplicated(cn)]
+  if (length(dup_col_names) != 0) {
+    Abort(paste0("Column name(s) ", 
+                 paste(unique(dup_col_names), 
+                       collapse = ", "), " must not be duplicated."))
+  } 
+  
+  # If not an epi_df (missing time_value or geo_value), return a tibble
+  not_epi_df <- (!("time_value" %in% cn) || !("geo_value" %in% cn))
```

Comment:
Style thing: remove outer parens?  (Inner parens may also be unnecessary, but still might help clarify.)

Reply:
Could you please add some test_that tests for the behaviors above?  I assume at least the duplicate-col error must be missing from the tests.

Reply:
Added tests for duplicate-col error and missing `geo_value` col.